internal/worker_deployment_client.go

```diff
@@ -0,0 +1,576 @@
+		RampingVersionPercentage float32
+
+		// CurrentVersionChangedTime - Last time the current version was changed.
+		CurrentVersionChangedTime time.Time
+
+		// RampingVersionChangedTime - Last time the ramping version was changed. Not updated if
+		// only RampingVersionPercentage changes.
+		RampingVersionChangedTime time.Time
+
+		// RampingVersionPercentageChangedTime - Last time ramping version percentage was changed.
+		// If RampingVersion has changed, this is also updated, even if the percentage remains the same.
+		RampingVersionPercentageChangedTime time.Time
+	}
+
+	// WorkerDeploymentListEntry is a subset of fields from [WorkerDeploymentInfo].
+	// NOTE: Experimental
+	//
+	// Exposed as: [go.temporal.io/sdk/client.WorkerDeploymentListEntry]
+	WorkerDeploymentListEntry struct {
+		// Name - The deployment name.
+		Name string
+
+		// CreateTime - When this deployment was created.
+		CreateTime time.Time
+
+		// RoutingConfig - When to execute new or existing Workflow Tasks with this Deployment.
+		RoutingConfig WorkerDeploymentRoutingConfig
+	}
+
+	// WorkerDeploymentListIterator is an iterator for deployments.
+	// NOTE: Experimental
+	//
+	// Exposed as: [go.temporal.io/sdk/client.WorkerDeploymentListIterator]
+	WorkerDeploymentListIterator interface {
+		// HasNext - Return whether this iterator has next value.
+		HasNext() bool
+
+		// Next - Returns the next Worker Deployment and error
+		Next() (*WorkerDeploymentListEntry, error)
+	}
+
+	// WorkerDeploymentDeleteOptions provides options for [WorkerDeploymentClient.Delete].
+	// NOTE: Experimental
+	//
+	// Exposed as: [go.temporal.io/sdk/client.WorkerDeploymentDeleteOptions]
+	WorkerDeploymentDeleteOptions struct {
+		// Name - The name of the deployment to be deleted.
+		Name string
+
+		// Identity - The identity of the client who initiated this request.
+		// Optional: defaulted to empty string
+		Identity string
+	}
+
+	// WorkerDeploymentDeleteResponse is the response for [WorkerDeploymentClient.Delete].
+	// NOTE: Experimental
+	//
+	// Exposed as: [go.temporal.io/sdk/client.WorkerDeploymentDeleteResponse]
+	WorkerDeploymentDeleteResponse struct {
+		// Placeholder for future extensions.
+	}
+
+	// WorkerDeploymentClient is the client that manages Worker Deployments.
+	// NOTE: Experimental
+	//
+	// Exposed as: [go.temporal.io/sdk/client.WorkerDeploymentClient]
+	WorkerDeploymentClient interface {
+		// List returns an iterator to enumerate Worker Deployments in the client's namespace.
+		// NOTE: Experimental
+		List(ctx context.Context, options WorkerDeploymentListOptions) (WorkerDeploymentListIterator, error)
+
+		// GetHandle returns a handle to a Worker Deployment.
+		//
+		// This method does not validate the Worker Deployment Name. If there is no deployment
+		// with that name in this namespace, methods like WorkerDeploymentHandle.Describe()
+		// will return an error.
+		// NOTE: Experimental
+		GetHandle(ctx context.Context, name string) WorkerDeploymentHandle
```

Comment:
```suggestion
		GetHandle(name string) WorkerDeploymentHandle
```

Not sure you need a context for this

Reply:
good point